### ISSUE TITLE
stateful client with a db

### ISSUE THREAD

**Issue by kanhegaonkarsaurabh** [kern/dongxi#1]
I've seen a TODO inside the source code that says that a stateful client needs to be implemented with a db for it. I'm just an avid user of things on my apple devices and loved this tool you've built. Is this issue something that maybe I could work on and contribute? Would a PR be acceptable for this issue?

**Comment by kern** [comment]
Thanks for reaching out! Glad you found this useful. The goal of the stateful client would be to:

- Persist all AORs/projects/tags/todo’s in a local database (i.e. PostgreSQL or SQLite)
- Automatically select the appropriate project ID based on name
- Support completing todo’s and checklists

I’m thinking of using something like TypeORM to accomplish this: https://github.com/typeorm/typeorm

If you’d like to get it started, would be happy to review! No need to do everything at once, happy to iterate with you, as I too want this functionality. :)

Sent via Superhuman iOS ( https://sprh.mn/?vip=alex@kern.io )

**Comment by kanhegaonkarsaurabh** [comment]
Yes this sounds good! I can definitely begin on looking into TypeORM and setting up a postgreSQL database for it. I’ll use this thread for further conversations then

**Comment by kern** [comment]
check out the repo now
